alonzo/impl/src/Cardano/Ledger/Alonzo/TxBody.hs

```diff
@@ -279,38 +290,77 @@ pattern TxBody
       )
   where
     TxBody
-      inputs'
-      inputs_fee'
-      outputs'
-      certs'
-      wdrls'
-      txfee'
-      vldt'
-      update'
-      mint'
-      sdHash'
-      adHash' =
+      inputsX
+      inputs_feeX
+      outputsX
+      certsX
+      wdrlsX
+      txfeeX
+      vldtX
+      updateX
+      mintX
+      sdHashX
+      adHashX =
         TxBodyConstr $
           memoBytes
             ( encodeTxBodyRaw $
                 TxBodyRaw
-                  inputs'
-                  inputs_fee'
-                  outputs'
-                  certs'
-                  wdrls'
-                  txfee'
-                  vldt'
-                  update'
-                  mint'
-                  sdHash'
-                  adHash'
+                  inputsX
+                  inputs_feeX
+                  outputsX
+                  certsX
+                  wdrlsX
+                  txfeeX
+                  vldtX
+                  updateX
+                  mintX
+                  sdHashX
+                  adHashX
             )
 
 {-# COMPLETE TxBody #-}
 
 instance (c ~ Crypto era, Era era) => HashAnnotated (TxBody era) EraIndependentTxBody c
 
+-- ==============================================================================
+-- We define these accessor functions manually, because if we define them using
+-- the record syntax in the TxBody pattern, they inherit the (AlonzoBody era)
+-- constraint as a precondition. This is unnecessary, as one can see below
+-- they need not be constrained at all. This should be fixed in the GHC compiler.
+
+inputs' :: TxBody era -> Set (TxIn (Crypto era))
```

Comment:
Same comment as before, that we don't actually need/use these functions.

Reply:
Ah, but we will.